server/sonar-server/src/main/java/org/sonar/server/util/cache/MemoryCache.java

```diff
@@ -54,12 +56,16 @@ public V getNullable(K key) {
   public V get(K key) {
     V value = getNullable(key);
     if (value == null) {
-      throw new IllegalArgumentException("Not found: " + key);
+      throw new NotFoundException("Not found: " + key);
     }
     return value;
   }
 
-  public Map<K, V> getAll(Iterable<K> keys) {
+  /**
+   * Get values associated with keys. All the requested keys are included
+   * in the Map result. Value is null if the key is not found in cache.
+   */
+  public Map<K, V> getAllNullable(Iterable<K> keys) {
```

Comment:
Go for getAll(). It's not only found in cache. Map contains all the keys, even if they are not found.